src/brogue/Movement.c

```diff
@@ -1056,6 +1056,10 @@ boolean playerMoves(short direction) {
                    && cellHasTerrainFlag(newX, newY, T_IS_DF_TRAP)
                    && !(pmap[newX][newY].flags & PRESSURE_PLATE_DEPRESSED)
                    && !cellHasTMFlag(newX, newY, TM_IS_SECRET)
+                   && (!rogue.armor || !(rogue.armor->flags & ITEM_RUNIC) || !(rogue.armor->flags & ITEM_RUNIC_IDENTIFIED) || rogue.armor->enchant2 != A_RESPIRATION ||
+                        (!cellHasTerrainType(newX, newY, GAS_TRAP_POISON)
```

Comment:
I was tempted to add and check for a new terrain flag along the lines of `T_IS_RESP_IMMUNE_DF_TRAP`, but opted to do the more explicit thing.

Reply:
This works :+1: